### ISSUE TITLE
Validation Error Spec for totalDigits and factionDigits

### ISSUE THREAD

**Issue by davlee1972** [brunato/xmlschema#5]
Leading zeros before the decimal point and trailing zeros after the decimal point should not be included in the validation check.

https://www.w3.org/TR/xmlschema-2/#rf-totalDigits
https://www.w3.org/TR/xmlschema-2/#rf-fractionDigits
"a lexical representation that adds additional leading zero digits or trailing fractional zero digits is still permitted".

`
<simpleType name='celsiusBodyTemp'>
  <restriction base='decimal'>
    <totalDigits value='4'/>
    <fractionDigits value='1'/>
  </restriction>
</simpleType>
`

Both of these values would raise validation errors when they shouldn't.
`
<celsiusBodyTemp>0500.0<celsiusBodyTemp>
<celsiusBodyTemp>500.00<celsiusBodyTemp>
`

**Comment by brunato** [comment]
Only the second test (eg `500.00`) raise errors for me (2 errors, both for totalDigits validator and for fractionDigits validator). 

Anyway I've found the problem. Has been corrected with the adding a `strip('0')` after in `str(x)` into facets validators (in module facets.py):

```
    def total_digits_validator(self, x):
        if len([d for d in str(x).strip('0') if d.isdigit()]) > self.value:
            yield XMLSchemaValidationError(self, x)

    def fraction_digits_validator(self, x):
        if len(str(x).strip('0').partition('.')[2]) > self.value:
            yield XMLSchemaValidationError(self, x)
```

Will be available soon and then in the next pip release (0.9.8).

**Comment by brunato** [comment]
Added a test for this. Fixed in the version 0.9.8.
